From 908e40dbafada45c5dbdd7998ea9bc982eae0a4f Mon Sep 17 00:00:00 2001
From: Ben Murdoch <benm@google.com>
Date: Mon, 9 Dec 2024 12:15:04 +0000
Subject: [PATCH] RESTRICT AUTOMERGE Remove flag check to sanitise keyboard
 shortcuts provided by apps.

Flag has been rolled out.

Bug: 331180422
Flag: EXEMPT removing com.android.systemui.validate_keyboard_shortcut_helper_icon_uri
Test: Manual
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:e1bee47e162d62984dd5f21686fed1cff19f6c21)
Merged-In: Iaca90a21f6337cdf7b149cf11aa7656cfb3bada4
Change-Id: Iaca90a21f6337cdf7b149cf11aa7656cfb3bada4
---
 .../systemui/statusbar/KeyboardShortcutListSearch.java |  8 ++------
 .../android/systemui/statusbar/KeyboardShortcuts.java  | 10 ++--------
 .../statusbar/KeyboardShortcutListSearchTest.java      |  4 ----
 .../systemui/statusbar/KeyboardShortcutsTest.java      |  4 ----
 4 files changed, 4 insertions(+), 22 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/KeyboardShortcutListSearch.java b/packages/SystemUI/src/com/android/systemui/statusbar/KeyboardShortcutListSearch.java
index c997ac5ad9df..134925111825 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/KeyboardShortcutListSearch.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/KeyboardShortcutListSearch.java
@@ -428,9 +428,7 @@ public final class KeyboardShortcutListSearch {
                 mKeySearchResultMap.put(SHORTCUT_SPECIFICAPP_INDEX, false);
             } else {
                 mCurrentAppPackageName = result.get(0).getPackageName();
-                if (validateKeyboardShortcutHelperIconUri()) {
-                    KeyboardShortcuts.sanitiseShortcuts(result);
-                }
+                KeyboardShortcuts.sanitiseShortcuts(result);
                 mSpecificAppGroup.addAll(
                         reMapToKeyboardShortcutMultiMappingGroup(result));
                 mKeySearchResultMap.put(SHORTCUT_SPECIFICAPP_INDEX, true);
@@ -446,9 +444,7 @@ public final class KeyboardShortcutListSearch {
         // Add specific Ime shortcuts
         if (result != null) {
             if (!result.isEmpty()) {
-                if (validateKeyboardShortcutHelperIconUri()) {
-                    KeyboardShortcuts.sanitiseShortcuts(result);
-                }
+                KeyboardShortcuts.sanitiseShortcuts(result);
                 mInputGroup.addAll(
                         reMapToKeyboardShortcutMultiMappingGroup(result));
             }
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/KeyboardShortcuts.java b/packages/SystemUI/src/com/android/systemui/statusbar/KeyboardShortcuts.java
index 766c391b14d8..a99da9db4b1a 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/KeyboardShortcuts.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/KeyboardShortcuts.java
@@ -412,10 +412,7 @@ public final class KeyboardShortcuts {
         mReceivedAppShortcutGroups =
                 result == null ? Collections.emptyList() : result;
 
-        if (validateKeyboardShortcutHelperIconUri()) {
-            sanitiseShortcuts(mReceivedAppShortcutGroups);
-        }
-
+        sanitiseShortcuts(mReceivedAppShortcutGroups);
         maybeMergeAndShowKeyboardShortcuts();
     }
 
@@ -423,10 +420,7 @@ public final class KeyboardShortcuts {
         mReceivedImeShortcutGroups =
                 result == null ? Collections.emptyList() : result;
 
-        if (validateKeyboardShortcutHelperIconUri()) {
-            sanitiseShortcuts(mReceivedImeShortcutGroups);
-        }
-
+        sanitiseShortcuts(mReceivedImeShortcutGroups);
         maybeMergeAndShowKeyboardShortcuts();
     }
 
diff --git a/packages/SystemUI/tests/src/com/android/systemui/statusbar/KeyboardShortcutListSearchTest.java b/packages/SystemUI/tests/src/com/android/systemui/statusbar/KeyboardShortcutListSearchTest.java
index 63e56eeb730f..8045a13ff9be 100644
--- a/packages/SystemUI/tests/src/com/android/systemui/statusbar/KeyboardShortcutListSearchTest.java
+++ b/packages/SystemUI/tests/src/com/android/systemui/statusbar/KeyboardShortcutListSearchTest.java
@@ -26,7 +26,6 @@ import static org.mockito.Mockito.when;
 
 import android.graphics.drawable.Icon;
 import android.os.Handler;
-import android.platform.test.annotations.EnableFlags;
 import android.view.KeyboardShortcutGroup;
 import android.view.KeyboardShortcutInfo;
 import android.view.WindowManager;
@@ -34,7 +33,6 @@ import android.view.WindowManager;
 import androidx.test.filters.SmallTest;
 import androidx.test.runner.AndroidJUnit4;
 
-import com.android.systemui.Flags;
 import com.android.systemui.SysuiTestCase;
 
 import com.google.android.material.bottomsheet.BottomSheetDialog;
@@ -95,7 +93,6 @@ public class KeyboardShortcutListSearchTest extends SysuiTestCase {
     }
 
     @Test
-    @EnableFlags(Flags.FLAG_VALIDATE_KEYBOARD_SHORTCUT_HELPER_ICON_URI)
     public void requestAppKeyboardShortcuts_callback_sanitisesIcons() {
         KeyboardShortcutGroup group = createKeyboardShortcutGroupForIconTests();
 
@@ -114,7 +111,6 @@ public class KeyboardShortcutListSearchTest extends SysuiTestCase {
     }
 
     @Test
-    @EnableFlags(Flags.FLAG_VALIDATE_KEYBOARD_SHORTCUT_HELPER_ICON_URI)
     public void requestImeKeyboardShortcuts_callback_sanitisesIcons() {
         KeyboardShortcutGroup group = createKeyboardShortcutGroupForIconTests();
 
diff --git a/packages/SystemUI/tests/src/com/android/systemui/statusbar/KeyboardShortcutsTest.java b/packages/SystemUI/tests/src/com/android/systemui/statusbar/KeyboardShortcutsTest.java
index 105cf168995c..20ecaf75c625 100644
--- a/packages/SystemUI/tests/src/com/android/systemui/statusbar/KeyboardShortcutsTest.java
+++ b/packages/SystemUI/tests/src/com/android/systemui/statusbar/KeyboardShortcutsTest.java
@@ -31,7 +31,6 @@ import android.annotation.Nullable;
 import android.app.Dialog;
 import android.graphics.drawable.Icon;
 import android.os.Handler;
-import android.platform.test.annotations.EnableFlags;
 import android.view.KeyboardShortcutGroup;
 import android.view.KeyboardShortcutInfo;
 import android.view.WindowManager;
@@ -39,7 +38,6 @@ import android.view.WindowManager;
 import androidx.test.filters.SmallTest;
 import androidx.test.runner.AndroidJUnit4;
 
-import com.android.systemui.Flags;
 import com.android.systemui.SysuiTestCase;
 
 import org.junit.Before;
@@ -131,7 +129,6 @@ public class KeyboardShortcutsTest extends SysuiTestCase {
     }
 
     @Test
-    @EnableFlags(Flags.FLAG_VALIDATE_KEYBOARD_SHORTCUT_HELPER_ICON_URI)
     public void requestAppKeyboardShortcuts_callback_sanitisesIcons() {
         KeyboardShortcutGroup group = createKeyboardShortcutGroupForIconTests();
         KeyboardShortcuts.toggle(mContext, DEVICE_ID);
@@ -143,7 +140,6 @@ public class KeyboardShortcutsTest extends SysuiTestCase {
     }
 
     @Test
-    @EnableFlags(Flags.FLAG_VALIDATE_KEYBOARD_SHORTCUT_HELPER_ICON_URI)
     public void requestImeKeyboardShortcuts_callback_sanitisesIcons() {
         KeyboardShortcutGroup group = createKeyboardShortcutGroupForIconTests();
         KeyboardShortcuts.toggle(mContext, DEVICE_ID);
-- 
2.34.1

